### PR TITLE
Documentation cleanup and minor refactor on the consensus package.

### PR DESCRIPTION
**Pull request by fnothaft**
* Moved `org.bdgenomics.adam.models.Concensus` to `org.bdgenomics.adam.algorithms.consensus.Consensus`.
* Made all classes in `org.bdgenomics.adam.algorithms.consensus` package private to `org.bdgenomics.adam`, along with `IndelTable` and `NormalizationUtils`.
* Fleshed out documentation for all classes and methods in `org.bdgenomics.adam.algorithms.consensus.Consensus`.
* Refactored `Consensus` method to use `ReferenceRegion` instead of `start`/`end` to check overlap bounds. This led to the discovery of an off by one error in `ConsensusSuite`, which was then fixed.

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/algorithms/consensus/Consensus.scala [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+   *
+   * Parses the read CIGAR and uses the read sequence to create consensuses. A
+   * consensus sequence is generated if an insertion or deletion is observed.
+   * When an insertion is observed, the insertion is cut from the read sequence
+   * to create the consensus.
+   *
+   * @note This method can only generate a single consensus from a read. Reads
+   *   with more than one INDEL operator are ignored.
+   *
+   * @param sequence Read sequence.
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/algorithms/consensus/Consensus.scala]: `@param` comments don't need to be full sentences

File: adam-core/src/main/scala/org/bdgenomics/adam/algorithms/consensus/Consensus.scala [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+   * @param start The start position of the read alignment.
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/algorithms/consensus/Consensus.scala]: e.g. I would prefer `@param start start position of the read alignment`

File: adam-core/src/main/scala/org/bdgenomics/adam/algorithms/consensus/Consensus.scala [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+   * @param cigar The CIGAR string for the local alignment of the read.
+   * @return Returns an optional consensus sequence if there is exactly one
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/algorithms/consensus/Consensus.scala]: `@return an optinoal consensus sequence...`

File: adam-core/src/main/scala/org/bdgenomics/adam/algorithms/consensus/Consensus.scala [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+ * @param index The reference region that this allele spans.
+ */
+private[adam] case class Consensus(consensus: String, index: ReferenceRegion) {
+
+  /**
+   * Inserts this consensus sequence into a reference genome sequence.
+   *
+   * @throws IllegalArgumentException If the consensus doesn't overlap with the
+   *   provided reference sequence.
+   *
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/algorithms/consensus/Consensus.scala]: `@throws IllegalArgumentException if the consensus sequence...`

File: adam-core/src/main/scala/org/bdgenomics/adam/algorithms/consensus/ConsensusGeneratorFromReads.scala [hunk 5]
```diff
@@ -68,7 +76,16 @@ class ConsensusGeneratorFromReads extends ConsensusGenerator {
...
-   * Generates concensus sequences from reads with indels.
+   * Generates concensus sequences from reads with INDELs.
+   *
+   * Loops over provided reads. Filters all reads without an MD tag, and then
+   * generates consensus sequences. If a read contains a single INDEL aligned to
+   * the reference, we emit that INDEL. Else, we do not emit a consensus from
+   * the read. We dedup the consensuses to remove any INDELs observed in
+   * multiple reads and return.
+   *
+   * @pararm Reads to search for INDELs.
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/algorithms/consensus/ConsensusGeneratorFromReads.scala]: typo

File: adam-core/src/main/scala/org/bdgenomics/adam/algorithms/consensus/ConsensusGeneratorFromReads.scala [hunk 6]
```diff
@@ -68,7 +76,16 @@ class ConsensusGeneratorFromReads extends ConsensusGenerator {
   }
 
   /**
-   * Generates concensus sequences from reads with indels.
+   * Generates concensus sequences from reads with INDELs.
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/algorithms/consensus/ConsensusGeneratorFromReads.scala]: `concensus` &rarr; `consensus`

**Comment by fnothaft** [comment]
Addressed typos and rebased.

**Comment by heuermh** [comment]
Thank you, @fnothaft!

**Pull request merged**